jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdAuthenticator.java

```diff
@@ -381,17 +401,29 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
         if (uri == null)
             uri = URIUtil.SLASH;
 
-        mandatory |= isJSecurityCheck(uri);
-        if (!mandatory)
-            return new DeferredAuthentication(this);
+        HttpSession session = request.getSession(false);
+        if (_openIdConfiguration.isRespectIdTokenExpiry() && hasExpiredIdToken(session))
```

Comment:
this reads better:
```suggestion
        if (hasExpiredIdToken(session) && _openIdConfiguration.isRespectIdTokenExpiry())
```

Reply:
Maybe, but its much cheaper to check the boolean than getting the claims from the session and verifying them.
So I think the current order is better.